extra/help/lint/coverage/coverage.factor

```diff
@@ -0,0 +1,146 @@
+            [ first2 write-object ] if
+        ] [ write ] if
+    ] each ; inline
+
+: (assemble-word-metadata) ( vec word -- vec )
+    [
+        [ "[" ] dip vocabulary>> dup lookup-vocab 2array "] "
+            3array over push-all
+    ] [
+        [ name>> ] keep 2array ": "
+        2array over push-all
+    ] bi ; inline
+
+: (assemble-empty-examples) ( vec coverage -- vec )
+    empty-examples?>> [ "empty " \ $examples [ name>> ] keep 2array "; "
+        3array over push-all
+    ] when ;
+
+: (assemble-omitted-sections) ( vec coverage -- vec )
+    omitted-sections>> [
+        length "section" ?pluralize ": " append
+    ] [
+        [ [ name>> ] keep 2array ] map
+    ] bi
+    [ "needs help " ] 2dip
+    3array over push-all ;
+
+: (present-coverage) ( coverage-report -- )
+    [ V{ } clone ] dip
+    [ word-name>> (assemble-word-metadata) ] keep
+    dup 100%-coverage?>> [ drop "full help coverage" over push ]
+    [ [ (assemble-empty-examples) ] [ (assemble-omitted-sections) ] bi ] if
+    "\n" over push
+    write-object-seq ; inline
+
+M: word-help-coverage summary
+    [ (present-coverage) ] with-string-writer ; inline
+
+: sorted-loaded-child-vocabs ( prefix -- assoc )
+    loaded-child-vocab-names natural-sort ; inline
+
+: resolve-name-in ( name namespaces -- word )
+    "syntax" swap remove " " join
+    "USING: " " ; \\ " surround
+    prepend eval( -- word ) ;
+
+: filter-private ( seq -- no-private )
+    [ ".private" ?tail nip not ] filter ; inline
+
+: ?pluralize ( n singular -- singular/plural )
+    count-of-things " " split1 nip ;
+
+: should-define ( word -- spec )
+    {
+        { [ dup predicate? ]    [ drop { } ] } ! predicate?s have generated docs
+        { [ dup error-class? ]  [ drop { $values $description $error-description } ] }
+        { [ dup class? ]        [ drop { $class-description } ] }
+        { [ dup generic? ]      [ drop { $values $contract $examples } ] }
+        { [ dup word? ]         [ drop { $values $description $examples } ] }
+        [ drop no-cond ]
+    } cond ;
+
+: word-defines-sections ( word -- seq )
+    word-help [ ignored-words member? not ] filter [ ?first ] map ;
+
+! only words that need examples, need to have them nonempty
+! not defining examples is not the same as an empty { $examples }
+: empty-examples? ( word -- ? )
+    word-help \ $examples swap elements [ f ] [ first rest empty? ] if-empty ;
+
+: missing-sections ( word -- missing )
+    [ should-define ] [ word-defines-sections ] bi diff ;
+PRIVATE>
+
+GENERIC: <word-help-coverage> ( word -- coverage )
+M: word <word-help-coverage>
+    dup [ missing-sections ] [ empty-examples? ] bi
+    2dup 2array { { } f } =
+    word-help-coverage boa ; inline
+
+M: string <word-help-coverage>
+    loaded-vocab-names resolve-name-in call-next-method ; inline
+
+: <vocab-help-coverage> ( vocab-spec -- coverage )
+    [ auto-use? off vocab-words natural-sort [ <word-help-coverage> ] map ] with-scope ;
```

Comment:
If you use my ``find-word`` above, then you can get rid of these ``with-scope`` stuff because there is no eval taking place.

Reply:
I take your point, but the `with-scope` here is because of `auto-use? off` and `group-articles vocab-articles set`, not because of `eval`. The `eval` would have the same behaviour regardless of the new namespace scope, because I am mimicking the definition of `help-lint` which looks like 

```factor
: help-lint ( prefix -- )
    [
        auto-use? off group-articles vocab-articles set
        loaded-child-vocab-names [ check-vocab ] each
    ] with-scope ;
```